stdlib/public/Darwin/Foundation/NSDictionary.swift

```diff
@@ -61,6 +61,78 @@ extension Dictionary : _ObjectiveCBridgeable {
                          to: NSDictionary.self)
   }
 
+  @inline(__always)
+  private static func _typePun<T>(index:Int, of buffer: UnsafePointer<AnyObject>, to: T.Type) -> UnsafeMutablePointer<T> {
+    let typedBase = UnsafeMutableRawPointer(mutating: buffer).assumingMemoryBound(to: T.self)
+    let unboundTarget = UnsafeMutableRawPointer(mutating: typedBase + index)
+    return unboundTarget.bindMemory(to: T.self, capacity: 1)
+  }
+
+  @inline(__always)
+  private static func _verbatimBridge<T>(
+    _ objectBuffer: UnsafePointer<AnyObject>,
+    count: Int,
+    to: T.Type
+  ) {
+    let verbatimBuffer = UnsafeMutableRawPointer(mutating: objectBuffer).assumingMemoryBound(to:T.self)
+    for i in (0..<count).reversed() {
+      _typePun(index: i, of: objectBuffer, to: T.self).initialize(to: verbatimBuffer[i])
+    }
+  }
+
+  @inline(__always)
+  private static func _nonVerbatimForceBridge<T>(
+    _ objectBuffer: UnsafePointer<AnyObject>,
+    count: Int,
+    to: T.Type
+  ) {
+    for i in (0..<count).reversed() {
+      let bridged = Swift._forceBridgeFromObjectiveC(objectBuffer[i], T.self)
+      _typePun(index: i, of: objectBuffer, to: T.self).initialize(to: bridged)
+    }
+  }
+  
+  @inline(__always)
+  private static func _nonVerbatimBridge<T>(
+    _ objectBuffer: UnsafePointer<AnyObject>,
+    count: Int,
+    to: T.Type
+  ) -> Bool {
+    for i in (0..<count).reversed() {
+      guard let bridged = objectBuffer[i] as? T else {
+        return false
+      }
+      _typePun(index: i, of: objectBuffer, to: T.self).initialize(to: bridged)
+    }
+    return true
+  }
+
+  @inline(__always)
+  private static func _forceBridge<T>(
+    _ objectBuffer: UnsafePointer<AnyObject>,
+    count: Int,
+    to: T.Type
+  ) {
+    if _isBridgedVerbatimToObjectiveC(T.self) {
+      _verbatimBridge(objectBuffer, count: count, to: T.self)
+    } else {
+      _nonVerbatimForceBridge(objectBuffer, count: count, to: T.self)
+    }
+  }
+
+  @inline(__always)
+  private static func _conditionallyBridge<T>(
+    _ objectBuffer: UnsafePointer<AnyObject>,
+    count: Int,
+    to: T.Type
+  ) -> Bool {
+    if _isBridgedVerbatimToObjectiveC(T.self) {
+      _verbatimBridge(objectBuffer, count: count, to: T.self)
+      return true
```

Comment:
This doesn't look right -- verbatim bridging should still perform a checked downcast, or we would allow casting an `NSDictionary` of `NSString` keys to `[NSNumber: Any]`. 

(Maybe there is room to add an unchecked downcasting variant for use in case we're totally sure.)

Reply:
Oops, it looks like we don't perform this check for verbatim-bridged forced downcasts on master, either -- we only check conditional downcasts. That seems like a bug!

Reply:
Except it isn't -- the check is of course still performed when we're downcasting individual items. This is by design; bridging is done lazily in these cases. Sorry for the brain hiccup!

Reply:
I *think* I have this right in the most recent update I pushed. The verbatim ones use ? and !, and the non verbatim ones use Swift._forceBridgeFromObjectiveC and Swift._conditionallyBridgeFromObjectiveC. It's possible we could merge some of the cases now, I'm not entirely sure.